sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EChangeFeedITest.scala

```diff
@@ -431,6 +528,25 @@ class SparkE2EChangeFeedITest
     rowsArray2 should have size 50 - initialCount
   }
 
+  private def compareArraysUnordered(inputArrayBuffer : ArrayBuffer[String], outputArray: Array[String]) : Boolean = {
+    println("Input : ", inputArrayBuffer.mkString(","))
+    println("Output : ", outputArray.mkString(","))
+    if (inputArrayBuffer.length != outputArray.length) {
+      return false
```

Comment:
Long story short - never use return in scala - this is one of the things in scala that confused the hack out of me initiially. return has different semantic based on from where the function is called - if you google for retrun and scala you'll find the details - long-story short - never use return, just amke sure the last line returns the value.

here to simplify I would simply rename the method to validateArraysUnordered and replcae the checks where you return false with an assert (the xxx shouldEqual yyy used in otherplaces)

Reply:
ack, will do.